TACC/TACC-Docs
# feat: docker-compose & readme

## Overview

Add `docker-compose.yml` and `README.md`.

## Testing

0. Workaround apparently missing files:

    ```
    ln -s ./helpsrc.md docs/include/help.md
    ln -s ./jobaccountingsrc.md docs/include/jobaccounting.md
    ln -s ./gromacssrc.md docs/softwareugs/gromacs.md
    touch docs/include/maverick2-jobaccounting.md
    ```

1. `docker-compose up`
2. Open https://localhost:8000/
3. Verify page loads with [ReadTheDocs theme](https://docs.readthedocs.io/en/stable/tutorial/) (not [MkDocs theme](https://www.mkdocs.org/getting-started/)).

## UI

### After `docker-compose build`

```
➜  TACC-Docs git:(task/add-docker-compose-and-readme) ✗ docker-compose up                                    
[+] Running 1/0
 ⠿ Container tup_docs  Created                                                                                                                                     0.0s
Attaching to tup_docs
tup_docs  | WARNING  -  Config value 'dev_addr': The use of the IP address '0.0.0.0' suggests a production environment or the use of a proxy to connect to the MkDocs server. However, the MkDocs' server is intended for local development purposes only. Please use a third party production-ready server instead.
tup_docs  | INFO     -  Building documentation...
tup_docs  | INFO     -  [macros] - Macros arguments: {'module_name': 'main', 'modules': [], 'include_dir': '', 'include_yaml': [], 'j2_block_start_string': '', 'j2_block_end_string': '', 'j2_variable_start_string': '', 'j2_variable_end_string': '', 'on_undefined': 'keep', 'on_error_fail': False, 'verbose': False}
tup_docs  | INFO     -  [macros] - Extra filters (module): ['pretty']
tup_docs  | INFO     -  Cleaning site directory
tup_docs  | INFO     -  The following pages exist in the docs directory, but are not included in the "nav" configuration:
tup_docs  |   - index.md
tup_docs  |   - aliases.md
tup_docs  |   - hpcugs/2mav/index.md
[...]
tup_docs  |   - tutorials/remotedesktopaccess.md
tup_docs  | INFO     -  Documentation built in 3.80 seconds
tup_docs  | INFO     -  [23:11:06] Watching paths for changes: '/code/docs', '/code/mkdocs.yml', '/opt/pysetup/.venv/lib/python3.10/site-packages/mkdocs/themes/readthedocs', '/opt/pysetup/.venv/lib/python3.10/site-packages/mkdocs/templates', '/opt/pysetup/.venv/lib/python3.10/site-packages/mkdocs/contrib/search/templates'
tup_docs  | INFO     -  [23:11:06] Serving on http://0.0.0.0:8000/
```

### Load the Website

![tacc-docs-docker-compose-running-site](https://user-images.githubusercontent.com/62723358/213033022-ec232722-8c80-492e-acf7-37cbc62c01e6.png)

## Known Issues

2. <details><summary>Production <code>docker-compose.yml</code> Configuration Warning</summary>

    ```
    WARNING  -  Config value 'dev_addr': The use of the IP address '0.0.0.0' suggests a production environment or the use of a proxy to connect to the MkDocs server. However, the MkDocs' server is intended for local development purposes only. Please use a third party production-ready server instead.
    ```

    We ignore this warning because production does not need nor use Docker (because it merely serves static files).

    </details>
